pyproject.toml

```diff
@@ -18,121 +18,100 @@ convention = "numpy"
-timm = "*"
-bitsandbytes = "*"
-optimum-quanto = ">=0.2.5"
-optimum = "*"
-ctranslate2 = "==4.5.0"
-whisper-s2t = "==1.3.0"
-hqq = "*"
-torchao = "*"
-llmcompressor = "*"
-# Added Optional Dependencies from Extras
-ruff = { version = "*", optional = true }
-jupyterlab = { version = "*", optional = true }
-notebook = { version = "*", optional = true }
-pre-commit = { version = "*", optional = true }
-build = { version = "*", optional = true }
-twine = { version = "*", optional = true }
-pyc-wheel = { version = "*", optional = true }
-pytest-cov = { version = "*", optional = true }
-coverage = { version = "*", optional = true }
-pytest = { version = "7.4.4", optional = true }
-docutils = { version = "*", optional = true }
-xformers = { version = "==0.0.30", optional = true }
-stable-fast-pruna = { version = "1.0.7", optional = true, markers = "extra == 'stable-fast' or extra == 'full'"}
-numpydoc-validation = { version = "*", optional = true }
-mypy = { version = "*", optional = true }
-types-PyYAML = { version = "*", optional = true }
-gptqmodel = { version = "==4.0.0.dev0+cu126torch2.7", optional = true , markers = "extra == 'gptq'"}
-logbar = { version = "*", optional = true, markers = "extra == 'gptq'" }
-tokenicer = { version = "*", optional = true, markers = "extra == 'gptq'" }
-threadpoolctl = { version = "*", optional = true, markers = "extra == 'gptq'" }
-device-smi = { version = "*", optional = true, markers = "extra == 'gptq'" }
-random-word = { version = "*", optional = true, markers = "extra == 'gptq'" }
+dependencies = [
+    "wget",
+    "torch==2.7.0",
+    "torchvision==0.22.0",
+    "torchmetrics[image]",
+    "requests>=2.31.0",
+    "transformers",
+    "pillow>=9.5.0",
+    "pytorch-lightning",
+    "huggingface-hub[hf-xet]>=0.30.0",
+    "datasets>=0.34",
+    "jsonschema",
+    "numpy>=1.24.4",
+    "setuptools",
+    "packaging",
+    "diffusers>=0.21.4",
+    "platformdirs",
+    "torch_pruning",
+    "python-dotenv",
+    "ConfigSpace>=1.2.1",
+    "cython",
+    "lago-python-client",
+    "sentencepiece",
+    "blobfile",
+    "librosa",
+    "openvino",
+    "soundfile",
+    "ipywidgets>=8.1.5",
+    "DeepCache",
+    "colorama",
+    "protobuf",
+    "peft",
+    "optuna",
+    "trl",
+    "opentelemetry-api>=1.30.0",
+    "opentelemetry-sdk>=1.30.0",
+    "opentelemetry-exporter-otlp>=1.29.0",
+    "codecarbon",
+    "pynvml",
+    "thop",
+    "timm",
+    "bitsandbytes",
+    "optimum-quanto>=0.2.5",
+    "optimum",
+    "ctranslate2==4.5.0",
+    "whisper-s2t==1.3.0",
+    "hqq",
+    "torchao",
+    "llmcompressor",
+]
 
-[tool.poetry.extras]
+[project.optional-dependencies]
 stable-fast = [
-    "xformers",
-    "stable-fast-pruna",
+    "xformers==0.0.30",
+    "stable-fast-pruna==1.0.7",
 ]
-# Note: You must first install the base package with ``pip install pruna`` before installing the GPTQ extension with ``pip install pruna[gptq]``
 gptq = [
-    "gptqmodel",
+    "gptqmodel==4.0.0.dev0+cu126torch2.7",
```

Comment:
I think this need to be more general

Reply:
I feel there was an issue and then @johnrachwan123 pinned this to resolve the issue, but it's coming from main branch

https://github.com/PrunaAI/pruna/blob/01a57b2ada6b56893fda0c232584955b7b8912ed/pyproject.toml#L115